mk/demo.mk

```diff
@@ -10,7 +10,7 @@ demo: rbd ## Expose RBD_IMAGE_NAME as NVMe-oF target
 	$(NVMEOF_CLI) create_bdev --pool $(RBD_POOL) --image $(RBD_IMAGE_NAME) --bdev $(BDEV_NAME)
 	$(NVMEOF_CLI) create_subsystem --subnqn $(NQN) --serial $(SERIAL)
 	$(NVMEOF_CLI) add_namespace --subnqn $(NQN) --bdev $(BDEV_NAME)
-	$(NVMEOF_CLI) create_listener --subnqn $(NQN) --trsvcid $(NVMEOF_IO_PORT)
+	$(NVMEOF_CLI) create_listener --subnqn $(NQN) --gateway-name $(NVMEOF_NAME) --traddr ${NVMEOF_IP_ADDRESS} --trsvcid $(NVMEOF_IO_PORT)
```

Comment:
Have you verified that the demo still works?  I'm asking because the gateway name isn't set in the default `ceph-nvmeof.conf` file and the gateway is supposed to pick up `socket.gethostname()` in that case.  However here `NVMEOF_NAME` would always be `ceph-nvmeof`.  Is that also the hostname reported by the container?

Reply:
I thought that if the name was incorrect, CI testing would detect it, but it didn't. Perhaps we could add `nvme discover -t tcp -a listener_ip -s port` checks in CI testing in the future. Now I have carefully reviewed the CI log and found that the gateway name is `nvmeof`, which is `hostname`?. But I don't know how to get the `hostname` in `demo.mk`, so I had to add a custom hostname. This is not a flexible method. Do you have any suggestions?

```
nvmeof_1                | INFO:control.server:Starting gateway nvmeof
...
nvmeof_1                | [2023-08-04 03:47:58.151150] tcp.c: 850:nvmf_tcp_listen: *NOTICE*: *** NVMe/TCP Target Listening on 192.168.13.3 port 4420 ***
nvmeof_1                | INFO:control.grpc:create_listener: True
```

Here print `create_listener: True`, it means correct.

Reply:
> Do you have any suggestions?

This is a question for @epuertat.

Reply:
@CongMinYin what's the exact requirement here, to have a customizable hostname (per-host or per-service instance)?

BTW, in the Dashboard we went through a lot of pain due to the reverse hostname look-ups (if you `strace` that, you'll see how many things the `gethostname` syscall tries to do to come up with system hostname)... so I'd be careful with that 🙈

Reply:
> @CongMinYin what's the exact requirement here, to have a customizable hostname (per-host or per-service instance)?
> 

@epuertat  We need to know the gateway name here. `create_listener` nedd require `-g gateway_name -a gateway_addr`. The default gateway name is host name.